### ISSUE TITLE
[Question][Hw7] Function Return type and return value type

### ISSUE THREAD

**Issue by KAIST-JongchanPark** [prosyslab-classroom/cs524-program-analysis#57]
When I type-check, I think I need to check "return type of current function" and "type of value which will be returned".

However, when I use below function and compile it, compiler add type-casting in .ll file automatically.

```
int f(int x) {
  char a = 0;
  return a;
}
```

Therefore, I modify .ll file manually but parser throws below exception.
`
Fatal error: exception Dummy_llvm_irreader.Error("test/example1.ll:15:7: error: value doesn't match function result type 'i32'\n  ret i8 %0\n      ^\n")
`

Because parser throws exception for this case, we don't need to consider such case?

**Comment by kangwoosukeq** [comment]
We will not modify any .ll files. You don't need to care about that.

**Comment by KAIST-JongchanPark** [comment]
However, when I remove test/example1.ll and recompile (by command `make example1.ll`) it, then It gives different .ll file from original one. 

Also, new example1.ll seems there is no type error.

In #51,  ShinWonho also said that example1.ll is slightly different from example1.c.

Could you check this?


------------------------------------------------------------------

Then, could you give an example .c file that causes type error after compile?

I made some .c files, but all of them don't cause type error after compile because compiler automatically add type-casting.

**Comment by kangwoosukeq** [comment]
I'm sorry to @KAIST-JongchanPark and everyone who saw this post. The `example1.ll` file is modified manually to cause type error. I delivered the wrong information in the first answer.
